docs/ingestion/compaction.md

```diff
@@ -0,0 +1,207 @@
+
+## Types of segment compaction
+You can configure the Druid Coordinator to perform automatic compaction, also called auto-compaction, for a datasource. Using a segment search policy, the coordinator periodically identifies segments for compaction starting with the newest to oldest. When segments can benefit from compaction, the coordinator automatically submits a compaction task. 
+
+Automatic compaction works in most use cases and should be your first option. To learn more about automatic compaction, see [Compacting Segments](../design/coordinator.md#compacting-segments).
+
+In cases where you require more control over compaction, you can manually submit compaction tasks. For example:
+- Automatic compaction is too slow.
+- You want to force compaction for a specific time range.
+- Compacting recent data before older data suboptimal is suboptimal in your environment.
+
+See [Setting up a manual compaction task](#setting-up-manual-compaction) more about manual compaction tasks.
+
+
+## Data handling with compaction
+During compaction, Druid overwrites the original set of segments with the compacted set without modifying the data. During compaction Druid locks the segments for the time interval being compacted to ensure data consistency.
+
+If an ingestion task needs to write data to a segment for a time interval locked for compaction, the ingestion task supersedes the compaction task and the compaction task fails without finishing. For manual compaction tasks you can adjust the input spec interval to avoid conflicts between ingestion and compaction. For automatic compaction, you can set the `skipOffsetFromLatest` key to adjustment the auto compaction starting point from the current time to reduce the chance of conflicts between ingestion and compaction. See [Compaction dynamic configuration](../configuration/index.md#compaction-dynamic-configuration) for more information.
+
+### Segment granularity handling
+
+Unless you modify the segment granularity in the [granularity spec](#compaction-granularity-spec), Druid attempts to retain the granularity for the compacted segments. When segments have different segment granularities with no overlap in interval Druid creates a separate compaction task for each to retain the segment granularity in the compacted segment. If segments have different segment granularities before compaction but there is some overlap in interval, Druid attempts find start and end of the overlapping interval and uses the closest segment granularity level for the compacted segment.
```

Comment:
This is unclear to me. Specifically, 
1. "segments have different segment granularities before compaction but there is some overlap in interval", I think an example can help.
2. "Druid attempts find start and end of the overlapping interval", I think more detail about why and how Druid do this attempt can help.
3. " uses the closest segment granularity level". Maybe an example here as well?

Reply:
@maytasm , I think by trying to clarify the behavior here, I made it more confusing. Can you help me with @2bethere 's questions?

Reply:
Thanks @maytasm , I will add your example here.